### DISCUSSION TITLE
Problem generating WSDL when having multiple services

### DISCUSSION THREAD

**Discussion by andreasdrake** [CoreWCF/CoreWCF#888] [General]
Hi!
I have a problem generating WSDL when I have two or more services like the code below.
When I go to https://localhost:7051/metadata only the first service (TestService1) is shown in the metadata.
I'm using CoreWCF 1.2. Am I doing something wrong here?

```
...
builder.Services.AddServiceModelServices().AddServiceModelMetadata();
builder.Services.AddSingleton<IServiceBehavior, UseRequestHeadersForMetadataAddressBehavior>();

WebApplication? app = builder.Build();
app.UseServiceModel(builder =>
{
    builder
        .AddService<TestService1>()
        .AddServiceEndpoint<TestService1, ITestService1>(new BasicHttpBinding(),
            "/TestService1");
	builder
        .AddService<TestService2>()
        .AddServiceEndpoint<TestService2, ITestService2>(new BasicHttpBinding(),
            "/TestService2");
});
var serviceMetadataBehavior = app.Services.GetRequiredService<ServiceMetadataBehavior>();
serviceMetadataBehavior.HttpsGetEnabled = true;
serviceMetadataBehavior.HttpsGetUrl = new Uri("https://localhost:7051/metadata");
...
```

**Comment by mconnew** [comment]
Each service will have it's own wsdl. The instance you get from DI allows you to modify options which apply to all services, but each service will get it's own copy. Use `ConfigureServiceHostBase` to modify it on a per service basis. If you don't set `HttpsGetUrl`, it will automatically listen at both https://localhost/TestService1 and https://localhost/TestService2, with each address only showing the wsdl for that service.  
There's basically a heuristic for deciding the address. If you have a configured base address (done by using the `AddService<TService>` overload which takes a delegate to configure ServiceOptions and then you can add the base address on the options class), it uses that. If you don't have a configured base address, then it goes through the endpoints in turn looking for one with a suitable url to hang off of.  
Although I'm not sure what will happen in your case as your endpoints are using HTTP (I think it doesn't care on the dispatch) and you want the WSDL to be on an HTTPS address. You might need to use a different constructor for BasicHttpBinding, e.g. `new BasicHttpBinding(BasicHttpSecurityMode.Transport)`. I can't remember how the logic for the heuristic functions when they don't match.
